src/pages-and-resources/app-settings-modal/AppSettingsModal.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import classNames from 'classnames';
+
+import { useDispatch } from 'react-redux';
+import {
+  Form, Button, TransitionReplace, ModalLayer, Hyperlink,
+} from '@edx/paragon';
+import FormSwitchGroup from '../../generic/FormSwitchGroup';
+import messages from './messages';
+
+import { enableDisableApp } from '../data/thunks';
+import { CoursePageShape } from '../pages/PageCard';
+import EnabledDisabledBadge from '../../generic/enabled-disabled-badge/EnabledDisabledBadge';
+import AppConfigFormDivider from '../discussions/app-config-form/apps/shared/AppConfigFormDivider';
+
+function AppSettingsModal({
+  intl, settings, title, children, onClose, onApply, enableAppLabel,
+  enableAppHelp, learnMoreURL, learnMoreText,
+}) {
+  const dispatch = useDispatch();
+  const learnMoreLink = (
+    <Hyperlink destination={learnMoreURL || 'https://www.google.com'} target="_blank">{learnMoreText}</Hyperlink>
+  );
+  return (
+    <ModalLayer
+      isOpen
+      closeText={intl.formatMessage(messages.cancel)}
+      dialogClassName="modal-dialog-centered modal-lg"
+      onClose={onClose}
+    >
+      <div
+        role="dialog"
+        aria-label={title}
+        className={classNames(
+          'bg-white',
+          'd-flex',
+          'flex-column',
+          'mw-xs',
+          'p-3',
+        )}
+      >
+        <h3>{title}</h3>
+        <AppConfigFormDivider />
+        <Form onSubmit={onApply}>
+          <FormSwitchGroup
+            id={`enable-${settings.id}-toggle`}
+            onChange={(e) => {
+              dispatch(enableDisableApp('', settings.id, e.target.value));
+            }}
+            onBlur={null}
+            defaultChecked={settings.isEnabled}
+            label={(
+              [
+                `${enableAppLabel} `,
+                <EnabledDisabledBadge enabled={settings.isEnabled} />,
+              ]
+            )}
+            helpText={
+              (
+                <p>
+                  {enableAppHelp}
+                  <br />
+                  {learnMoreLink}
+                </p>
+              )
+            }
+          />
+          <AppConfigFormDivider />
+          <TransitionReplace>
+            {settings.isEnabled
+              ? <React.Fragment key="app-enabled">{children}</React.Fragment>
+              : <React.Fragment key="app-disabled" />}
+          </TransitionReplace>
+        </Form>
+        {settings.isEnabled && children && <AppConfigFormDivider />}
+        <div className="d-flex justify-content-end">
+          <Button variant="cancel" className="btn btn-link" onClick={onClose}>
+            {intl.formatMessage(messages.cancel)}
+          </Button>
+          <Button
+            variant="success"
+            data-autofocus
+            onClick={() => {
+              if (onApply) { onApply(); }
+              onClose();
+            }}
```

Comment:
it should have input type= submit and it will automatically use onSubmit of form. why are we passing onClick action in this button?

Reply:
For that, it will need to be linked to that form, which it is not, and AFAIK it isn't straightforward with react.